clicktests/helpers.js

```diff
@@ -4,44 +4,66 @@ helpers.log = function(text) {
   console.log((new Date()).toISOString(), text);
 }
 
-helpers.waitForElement = function(page, selector, callback) {
-  var tryFind = function() {
-    helpers.log('Trying to find element: ' + selector);
-    var element = page.evaluate(function(selector) {
-      return document.querySelector(selector);
-    }, selector);
-    if (element) {
-      helpers.log('Found element: ' + selector);
-      callback(element);
-    }
-    else setTimeout(tryFind, 500);
+helpers.elementExists = function(page, selector) {
+  helpers.log('Querying element exists: ' + selector);
+  var element = page.evaluate(function(selector) {
+    return document.querySelector(selector);
+  }, selector);
+  if (element) {
+    helpers.log('Element exists: ' + selector);
+  } else {
+    helpers.log('Element doesn\'t exist: ' + selector);
   }
-  tryFind();
+  return element;
+}
+
+helpers.elementVisible = function(page, selector) {
+  helpers.log('Querying element visible: ' + selector);
+  var element = page.evaluate(function(selector) {
+    var element = document.querySelector(selector);
+    if (!element) return null;
+    var rect = element.getBoundingClientRect();
+    if (rect.width == 0 || rect.height == 0) return null;
```

Comment:
Should we add [`:visible`](https://stackoverflow.com/questions/8774089/detect-if-an-element-is-visible) to the selector?  I think that will capture when `display: none` case.